core/foundation/inc/RtypesCore.h

```diff
@@ -34,91 +45,91 @@ class TRootIOCtor;
+typedef char           Text_t;      ///< General string (char) \deprecated Consider replacing with `char`.
+typedef bool           Bool_t;      ///< Boolean (0=false, 1=true) (bool) \deprecated Consider replacing with `bool`.
+typedef unsigned char  Byte_t;      ///< Byte (8 bits) (unsigned char) \deprecated Consider replacing with `unsigned char` or `std::byte`.
+typedef short          Version_t;   ///< Class version identifier (short) \deprecated Consider replacing with `short` or `std::int16_t`.
+typedef const char     Option_t;    ///< Option string (const char) \deprecated Consider replacing with `const char`.
+typedef int            Ssiz_t;      ///< String size (int) \deprecated Consider replacing with `int`.
+typedef float          Real_t;      ///< TVector and TMatrix element type (float) \deprecated Consider replacing with `float`.
+typedef long long          Long64_t; ///< Portable signed long integer 8 bytes \deprecated Consider replacing with `long long` or `std::int64_t`.
+typedef unsigned long long ULong64_t;///< Portable unsigned long integer 8 bytes \deprecated Consider replacing with `unsigned long long` or `std::uint64_t`.
 #ifdef _WIN64
-typedef long long      Longptr_t;   //Integer large enough to hold a pointer
-typedef unsigned long long ULongptr_t;  //Unsigned integer large enough to hold a pointer
+typedef long long      Longptr_t;     ///< Integer large enough to hold a pointer \deprecated Consider replacing with `long long` on Windows.
+typedef unsigned long long ULongptr_t;///< Unsigned integer large enough to hold a pointer \deprecated Consider replacing with `unsigned long long` on Windows.
 #else
-typedef long           Longptr_t;   //Integer large enough to hold a pointer
-typedef unsigned long  ULongptr_t;  //Unsigned integer large enough to hold a pointer
+typedef long           Longptr_t;   ///< Integer large enough to hold a pointer \deprecated Consider replacing with `long` on UNIX.
+typedef unsigned long  ULongptr_t;  ///< Unsigned integer large enough to hold a pointer \deprecated Consider replacing with `unsigned long` on UNIX.
 #endif
-typedef double         Axis_t;      //Axis values type (double)
-typedef double         Stat_t;      //Statistics type (double)
-
-typedef short          Font_t;      //Font number (short)
-typedef short          Style_t;     //Style number (short)
-typedef short          Marker_t;    //Marker number (short)
-typedef short          Width_t;     //Line width (short)
-typedef short          Color_t;     //Color number (short)
-typedef short          SCoord_t;    //Screen coordinates (short)
-typedef double         Coord_t;     //Pad world coordinates (double)
-typedef float          Angle_t;     //Graphics angle (float)
-typedef float          Size_t;      //Attribute size (float)
+typedef double         Axis_t;      ///< Axis values type (double) \deprecated Consider replacing with `double`
```

Comment:
Replacing this series of types with their typedefs makes the code less readable by removing the semantic part of the type .... I do not think we should deprecated them.